src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *                       type: array
+ *                       items:
+ *                         type: string
+ *                         description: List of user IDs who liked the post.
+ *                 msg:
+ *                   type: string
+ *                   description: Status message indicating whether the post was liked or unliked.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 message:
+ *                   type: string
+ *                   description: Error message.
+ */
 router.post('/likePost/:postID', auth, likePost);
+
+/**
+ * @swagger
+ * /posts/singleImgDel/{postID}/{imgID}:
+ *   delete:
+ *     tags:
+ *       - Posts operations
+ *     summary: Delete an image from a post , Must be connected as user
+ *     description: Removes a specific image from a post and deletes the image from Cloudinary using its image ID. The post's image list is updated accordingly.
+ *     security:
+ *       - BearerAuth: []  # Indicates that the endpoint requires a bearer token for authentication
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post from which the image will be deleted.
+ *       - in: path
+ *         name: imgID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the image to be deleted from Cloudinary.
+ *     responses:
+ *       200:
+ *         description: Success message indicating that the image was deleted from both the post and Cloudinary.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Status message confirming the successful deletion of the image.
+ *                 error:
+ *                   type: object
+ *                   description: Contains error details if an error occurred during image deletion from Cloudinary.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating that an error occurred.
+ *                 error:
+ *                   type: object
+ *                   description: Details of the error encountered.
+ */
 router.post('/singleImgDel/:postID/:imgID', auth, deleteSinglePostImage);
+
+/**
+ * @swagger
+ * /cancel-delete:
+ *   post:
+ *     tags:
+ *       - General operations
+ *     summary: Handle cancel delete operation, Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"